chasm/lib/callback/component.go

```diff
@@ -1,22 +1,49 @@
 package callback
 
 import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"net/http/httptrace"
+	"slices"
+	"time"
+
+	"github.com/nexus-rpc/sdk-go/nexus"
 	"go.temporal.io/server/chasm"
 	callbackspb "go.temporal.io/server/chasm/lib/callback/gen/callbackpb/v1"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/namespace"
+	commonnexus "go.temporal.io/server/common/nexus"
+	"go.temporal.io/server/common/nexus/nexusrpc"
+	"go.temporal.io/server/service/history/queues"
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 const (
 	Archetype chasm.Archetype = "Callback"
 )
 
+// Callback represents a callback component in CHASM.
+//
+// This is the CHASM port of HSM's nexusInvocation struct from nexus_invocation.go:25-32.
 type Callback struct {
 	chasm.UnimplementedComponent
 
 	// Persisted internal state
 	*callbackspb.CallbackState
 
-	status *callbackspb.CallbackStatus
+	// Interface to retrieve Nexus operation completion data
+	CanGetNexusCompletion chasm.Field[CanGetNexusCompletion]
+
+	// Fields from HSM's nexusInvocation struct (nexus_invocation.go:35-40)
+	// These hold the invocation context needed for the HTTP request
+	completion nexusrpc.OperationCompletion
```

Comment:
Don't hold this here, create a separate struct for that as we did in the HSM version.